ddtrace/llmobs/_llmobs.py

```diff
@@ -1768,50 +1804,64 @@ def inject_distributed_headers(cls, request_headers: Dict[str, str], span: Optio
                 error = "no_active_span"
-                log.warning("No span provided and no currently active span found.")
-                return request_headers
+                raise LLMObsInjectDistributedHeadersError("No span provided and no currently active span found.")
             if not isinstance(span, Span):
-                error = "invalid_span"
-                log.warning("span must be a valid Span object. Distributed context will not be injected.")
-                return request_headers
+                raise LLMObsInjectDistributedHeadersError(
+                    "span must be a valid Span object. Distributed context will not be injected."
+                )
             HTTPPropagator.inject(span.context, request_headers)
             return request_headers
         finally:
             telemetry.record_inject_distributed_headers(error)
 
     @classmethod
-    def _activate_llmobs_distributed_context(cls, request_headers: Dict[str, str], context: Context) -> Optional[str]:
-        if cls.enabled is False:
-            return None
-        if not context.trace_id or not context.span_id:
-            log.warning("Failed to extract trace/span ID from request headers.")
-            return "missing_context"
-        _parent_id = context._meta.get(PROPAGATED_PARENT_ID_KEY)
-        if _parent_id is None:
-            log.debug("Failed to extract LLMObs parent ID from request headers.")
-            return "missing_parent_id"
+    def _activate_llmobs_distributed_context_soft_fail(cls, request_headers: Dict[str, str], context: Context) -> None:
+        cls._activate_llmobs_distributed_context(request_headers, context, _soft_fail=True)
+
+    @classmethod
+    def _activate_llmobs_distributed_context(
+        cls, request_headers: Dict[str, str], context: Context, _soft_fail: bool = False
+    ) -> None:
+        error = None
         try:
-            parent_id = int(_parent_id)
-        except ValueError:
-            log.warning("Failed to parse LLMObs parent ID from request headers.")
-            return "invalid_parent_id"
-        parent_llmobs_trace_id = context._meta.get(PROPAGATED_LLMOBS_TRACE_ID_KEY)
-        if parent_llmobs_trace_id is None:
-            log.debug("Failed to extract LLMObs trace ID from request headers. Expected string, got None.")
+            if cls.enabled is False:
+                return
+            if not context.trace_id or not context.span_id:
+                error = "missing_context"
+                if _soft_fail:
+                    log.warning("Failed to extract trace/span ID from request headers.")
+                    return
+                raise LLMObsActivateDistributedHeadersError("Failed to extract trace/span ID from request headers.")
+            _parent_id = context._meta.get(PROPAGATED_PARENT_ID_KEY)
+            if _parent_id is None:
+                error = "missing_parent_id"
+                log.debug("Failed to extract LLMObs parent ID from request headers.")
```

Comment:
nit: Should w log the context object here? It could make this logic easier to debug in customer environments.